pkg/kubectl/cmd/edit.go

```diff
@@ -167,14 +169,22 @@ func RunEdit(f *cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args
 		return err
 	}
 
-	infos, err := r.Infos()
+	clientConfig, err := f.ClientConfig()
 	if err != nil {
 		return err
 	}
 
-	clientConfig, err := f.ClientConfig()
+	infos := []*resource.Info{}
+	allErrs := []error{}
+	err = r.Visit(func(info *resource.Info, err error) error {
```

Comment:
Good catch - I was being creative on a previous attempt at getting things to work and I didn't realize I left the code to basically recreate `r.Infos()`, rather than enhancing it like I've done with other portions of the recursive feature.

I'll update it to use `r.Infos()`. Thanks